### PR TITLE
Move Wayland info to README and add disclaimer to AppInfo

### PR DESCRIPTION
**Pull request by elsiehupp**
Including a disclaimer might make the official Element maintainers be more inclined to [acknowledge](https://github.com/vector-im/element.io/issues/18) that this distribution exists. (Actually “adopting” this distribution and helping to [fix](https://github.com/vector-im/element-web/issues/3718) some of the known issues of the Flatpak does not seem to be a priority for them, though.)

@vector-im tagging you in case you want to weigh in the inclusion of a disclaimer.

### CODE REVIEW

File: im.riot.Riot.appdata.xml [hunk 1]
```diff
@@ -3,10 +3,13 @@
   <id>im.riot.Riot.desktop</id>
   <name>Element</name>
   <project_license>Apache-2.0</project_license>
-  <developer_name>New Vector Ltd</developer_name>
+  <developer_name>Element Software, Inc.</developer_name>
```
Comment (aaronraimist) [inline im.riot.Riot.appdata.xml:6]: New Vector Ltd is still correct AFAIK.

Comment (elsiehupp) [reply im.riot.Riot.appdata.xml:6]: I clicked through [the citation on the Wikipedia page](https://element.io/copyright), and New Vector Ltd is d.b.a. “Element”, and the US subsidiary is indeed called Element Software Inc. I used the latter because it was clearer for the purpose of the disclaimer I added (i.e. I wouldn’t have to say “New Vector Ltd, the maintainers of Element” or something similarly wordy), and I changed the `<developer_name/>` for consistency, even if using the name of the US subsidiary (rather than the d.b.a., which is also confusingly the name of the application itself) isn’t strictly correct.

File: im.riot.Riot.appdata.xml [hunk 2]
```diff
@@ -3,10 +3,13 @@
...
   <summary>Create, share, communicate, chat and call securely, and bridge to other apps</summary>
   <metadata_license>CC0-1.0</metadata_license>
   <url type="homepage">https://element.io/</url>
+  <url type="faq">https://github.com/flathub/im.riot.Riot/</url>
```
Comment (SISheogorath) [inline im.riot.Riot.appdata.xml:10]: I don't think this repository makes any sense to be linked as FAQ. If at all, this should be elements help page, but that's already linked below.

File: im.riot.Riot.appdata.xml [hunk 3]
```diff
@@ -16,9 +19,7 @@
       <li>Grab the attention of your colleague by calling out their name and don’t miss a thing with keyword alerts.</li>
       <li>Deploy bots for fun or practical use with our integrations store.</li>
     </ul>
-    <p>Preliminary Wayland support since 1.7.25.</p>
-    <p>To try running Element natively under Wayland, run: <code>flatpak run --socket=wayland --enable-features=UseOzonePlatform,WebRTCPipeWireCapturer --ozone-platform=wayland im.riot.Riot</code></p>
-    <p>For GNOME, window decorations are currently missing and you'll have to use keyboard shortcuts instead to resize the window.</p>
+    <p>NOTE: the Flathub distribution of Element is community maintained and not affiliated with or officially supported by Element Software, Inc. The Flathub distribution of Element has several known known issues, including broken functionality. To install the officially supported version of Element (Debian/Ubuntu only), visit the page labeled "Homepage". For known issues and experimental features, visit the pages labeled "Bugtracker" and "FAQ". Please do not report bugs to the upstream Element maintainers unless you can consistently reproduce them on the official distribution.</p>
```
Comment (SISheogorath) [inline im.riot.Riot.appdata.xml:22]: Honestly speaking, I'm not a fan. We don't put disclaimers on distribution packages, so why here?

Also, the set of bugs that are flatpak specific is rather limited. Main complaints are about the filesystem access being rather limited, which is intentional until electron figures out portals. The concern about encryption, as we figured out in the element packager chat, is not unique to the flatpak but is experienced by other distro packages as well. So for the most part, we actually want people to work with upstream, not with us.

**Review by SISheogorath — COMMENTED** [review]
Let me put it this way: I'm not really a fan of this idea.

**Comment by SISheogorath** [comment]
> Including a disclaimer might make the official Element maintainers be more inclined to acknowledge that this distribution exists. 

Just as hint, the element maintainers are well aware of this distribution as we are part of their Package Maintainers program.

**Comment by elsiehupp** [comment]
> Honestly speaking, I'm not a fan. We don't put disclaimers on other distribution packages, so why here?

This is [an open issue](https://github.com/flathub/linux-store-frontend/issues/214) with the way Flathub displays the information. I encourage you to read (and possibly respond) to that issue with regard to disclaimers.

My impression is that it is not unusual for third-party packagers on Flatpak to attach disclaimers. For example, see [Visual Studio Code](https://flathub.org/apps/details/com.visualstudio.code), [Minecraft](https://flathub.org/apps/details/com.mojang.Minecraft), [Minecraft Bedrock Launcher](https://flathub.org/apps/details/io.mrarm.mcpelauncher), [Steam](https://flathub.org/apps/details/com.valvesoftware.Steam), [Discord](https://flathub.org/apps/details/com.discordapp.Discord), [Zoom](https://flathub.org/apps/details/us.zoom.Zoom), [Microsoft Teams](https://flathub.org/apps/details/com.microsoft.Teams), etc. (I’m just scrolling through the “Popular” list on the Flathub website; these examples are not difficult to find.)

Flathub differs from distro-run package repositories in that packaging on distro repositories is somewhat more centrally supervised, whereas Flathub adopts the same aesthetic of first-party packaging as the Apple App Store and the Google Play Store. Indeed, some of the spicier comments on the [open issue](https://github.com/flathub/linux-store-frontend/issues/214) suggest that the existing presentation could even be considered misleading.

> Also, the set of bugs that are flatpak specific is rather limited. Main complaints are about the filesystem access being rather limited, which is intentional until electron figures out portals. The concern about encryption, as we figured out in the element packager chat, is not unique to the flatpak but is experienced by other distro packages as well. So for the most part, we actually want people to work with upstream, not with us.

I made this pull request directly in response to @aaronraimist’s comment on the issue I opened on the element.io repository: https://github.com/vector-im/element.io/issues/18

Basically my thought was that if the relationship between the Flatpak maintainers and the upstream Element maintainers was more explicitly elaborated on the Flathub page, they might be more inclined to list it on [the Element.io “Get Started” page](https://element.io/get-started).

I tagged @vector-im because this isn’t really my negotiation to have, but maybe I should tag @vector-im/element.io? I’m not exactly sure how tagging works. I would also encourage @aaronraimist to engage with you, and I would encourage you (@SISheogorath) to engage with [the issue regarding the element.io website](https://github.com/vector-im/element.io/issues/18).

> > Including a disclaimer might make the official Element maintainers be more inclined to acknowledge that this distribution exists.
> 
> Just as hint, the element maintainers are well aware of this distribution as we are part of their Package Maintainers program.

You removed the hyperlink on the word “acknowledge” in your quote, which directed specifically to the issue I’ve repeatedly linked to in this comment. They may have an official relation with you, but they don’t actually direct end users to download your package. They don’t actually direct end users to download *any* third-packages, which is kind of a problem from a user-onboarding standpoint, relative to how they present iOS, Android, macOS, and Windows, as you can see in this screenshot:

Element io download screenshot

FYI, the reason I moved the Wayland thing to a README was, aside from brevity, the fact that the string inside the `<code></code>` tags doesn’t actually render [on the Flathub website](https://flathub.org/apps/details/im.riot.Riot):

missing code section

(I couldn’t see how it looks in GNOME Software, though.) The way Flathub renders AppStream data just isn’t conducive to command-line tweaks. If it’s important to keep that, you could open an issue on [the Flathub website issue tracker](https://github.com/flathub/linux-store-frontend/issues) asking them to add support for `<code></code>` tags.

…

The shell-command rendering bug aside, I do think it’s important to acknowledge the specific limitations of the Flatpak distribution on the Flathub description page. And from the standpoint of maintaining some degree of brevity in the Flathub description page, it makes sense to have an external page (e.g. the README.md) to allow further elaboration.

I get that my wording is a bit ominous, and I’d be happy to change it. It would be most helpful if you, @aaronraimist, or any of the @vector-im maintainers could give specific feedback and guidance on the best way of doing so.

**Comment by aaronraimist** [comment]
Just to be clear, I am not part of Element. i am just another community member.

I previously raised https://github.com/flathub/im.riot.Riot/issues/87 and we basically came to the conclusion that it is a Flathub issue that there isn't a great way to display if a package is maintained by the community or by the upstream developers. Since there is not a specific field for that, I don't think would be a bad idea to add a little note to the end of the description on Flathub saying it is community maintained but I understand the hesitance to do so.

> I get that my wording is a bit ominous, and I’d be happy to change it.

@elsiehupp Yeah you might want to just shorten it to

> This is a community maintained package

or similar, rather than talking about issues with the package.

**Comment by elsiehupp** [comment]
Yeah, reading through #87 it seems like it's part of the broader issue of Flathub adopting the aesthetic of the Apple App Store, which implicitly changes people's expectations.

**Comment by SISheogorath** [comment]
> If it’s important to keep that, you could open an issue on the Flathub website issue tracker asking them to add support for `<code></code>` tags.

Yes, this is a know issue and not even unique to flatpak as it turned out. It's a problem with the c-library used to parse the description which, despite the Spec allowing code- and em-tags, simply strips them always entirely. Either way, yes, we should fix that, thanks for the reminder, I'll do this later today.

> My impression is that it is not unusual for third-party packagers on Flatpak to attach disclaimers. For example, see Visual Studio Code, Minecraft, Minecraft Bedrock Launcher, Steam, Discord, Zoom, Microsoft Teams, etc. (I’m just scrolling through the “Popular” list on the Flathub website; these examples are not difficult to find.)

Yes, indeed it's not unpopular for proprietary apps. But this lies in their nature and how flatpak distributes them, as flatpak itself doesn't redistribute any binary but only instructs the local flatpak client to download the binary from the vendor page.

But to this day this is still up to the maintainer, otherwise this would be done by the store-frontend.

---

To me this PR seems to try to push the flatpak towards a more official status, but ignores that the element developers are well aware of it, and decided not to. And again, I would point to other distribution packages, which are also not mentioned anywhere while still being actively supported by vector.

Maybe I just don't share your vision of "This flatpak needs to be more known/recognised".

**Comment by elsiehupp** [comment]
Yes, the underlying problem (as I laid out in the other issue I linked) is that the Element app onboarding process for Linux is significantly less user friendly than the Element app onboarding process for any of the other operating systems Element supports. Either you have to use the command line, or you get torpedoed with undisclosed broken functionality on Flatpak. Neither option is great.

(And, yes, I know how to use the command line. But a big part of why Element exists is to market Matrix as a turn-key solution.)

**Comment by Erick555** [comment]
Adding a line about unofficial status in appdata is ok if upstream request it but advertising _broken functionality_ and pointing people to alternative installations looks rather bizarre.

File: README.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+For known issues, visit the [Issues](https://github.com/flathub/im.riot.Riot/issues) page above. Please do not report bugs to the upstream Element maintainers unless you can consistently reproduce them on the official distribution.
+
+## Experimental Features
+
+### Preliminary Wayland support since 1.7.25.
+
+To try running Element natively under Wayland, run:
+```bash
+$ flatpak run --socket=wayland --enable-features=UseOzonePlatform,WebRTCPipeWireCapturer --ozone-platform=wayland im.riot.Riot
```
Comment (TheEvilSkeleton) [inline README.md:25]: I get the following error when I type the current command:

```
❯ flatpak run --socket=wayland --enable-features=UseOzonePlatform,WebRTCPipeWireCapturer --ozone-platform=wayland im.riot.Riot
error: Unknown option --enable-features=UseOzonePlatform,WebRTCPipeWireCapturer
```

The working workaround/fix is:

```suggestion
$ flatpak run im.riot.Riot --socket=wayland --enable-features=UseOzonePlatform,WebRTCPipeWireCapturer --ozone-platform=wayland
```

**Comment by barthalion** [comment]
As neither of you is in any way related to Vector, and Element itself is perfectly redistributable, there's nothing to do here.

**Comment by elsiehupp** [comment]
I mean, that's fine. The changes I was suggesting here are entirely incidental to the issue I opened on the element.io website issue tracker regarding Linux onboarding. I thought a disclaimer here might facilitate solving that issue, but it's entirely possible that that isn't the case. 🤷🏻‍♀️

**Comment by aaronraimist** [comment]
@barthalion I mean the CEO of Element did say they would like note to be added saying this package is not maintained by Element https://github.com/flathub/im.riot.Riot/issues/87#issuecomment-634351030

**Comment by barthalion** [comment]
@aaronraimist Which is fine, but should be solved on project-wide basis and not discussed under every app. As we want to be accommodating, we usually agree for adding a disclaimer, but technically speaking Element is available under permissive license, and I doubt Vector employees are reaching out to every single distributor shipping their software,

**Comment by elsiehupp** [comment]
@aaronraimist would you be up for opening your own pull request, then? I kind of don't want to be responsible for coming up with precise wording that everyone will be happy with.

FWIW my understanding with redistributing branded libre software is that the issue isn't the copyright; the issue is the trademark. While Element (the application) is GPL or whatever, Element (the name and logo) are protected.

IANAL, but referring to a trademarked entity (e.g. a software application) is acceptable inasmuch as the entity actually is the entity to which the trademark owner applies the trademark. If the entity is not the same entity (e.g. the software application is a modified fork), then the distributor would need explicit permission from the trademark owner to use the trademark to refer to the modified entity.

tl;dr if the version of Element distributed via Flathub is not 100% identical to the version of Element distributed by New Vector Ltd, then Flathub needs the permission of New Vector Ltd to call their version "Element".

(I can't personally attest to whether any substantial differences exist between the first-party binary distributed by New Vector Ltd and the version available here, but it sounds like certain stakeholders do have opinions on the matter.)

Again, my goal here is for Element's official "Get Started" instructions for Linux to be as user-friendly as its installation instructions for Android, iOS, macOS, and Windows, and accomplishing that goal requires cooperation with and the goodwill of the people who control the "Get Started" instructions on Element.io. If the folks at New Vector Ltd want a disclaimer, then it would be best to give them what the want.

**Comment by aaronraimist** [comment]
Well it seems clear the Flathub folks don't want to add a note about it being community maintained so I'm not going to try to push the issue further.

**Comment by Erick555** [comment]
> tl;dr if the version of Element distributed via Flathub is not 100% identical to the version of Element distributed by New Vector Ltd, then Flathub needs the permission of New Vector Ltd to call their version "Element".

The version of Element distributed via Flathub **is 100% identical** to the version of Element distributed by New Vector Ltd. You can read from manifest that it downloads official Element deb package from upstream server and copy all of its contents to flatpak location.

**Comment by barthalion** [comment]
@elsiehupp There's plenty of nuances to law, and here it falls under nominative fair use, so we are not abusing the copyright here; and if we were, no "it's not affiliated" disclaimer would fix that.

**Comment by elsiehupp** [comment]
Cool! Thank you for the explanation.
